core/src/main/kotlin/com/ritense/valtimo/camunda/domain/CamundaExecution.kt

```diff
@@ -27,78 +27,81 @@ import jakarta.persistence.ManyToOne
 import jakarta.persistence.OneToMany
 import jakarta.persistence.Table
 import jakarta.persistence.Transient
+import org.hibernate.annotations.Immutable
 
+@Immutable
 @Entity
 @Table(name = "ACT_RU_EXECUTION")
 @JsonIdentityInfo(generator = ObjectIdGenerators.PropertyGenerator::class, property = "id")
 class CamundaExecution(
 
     @Id
-    @Column(name = "ID_")
+    @Column(name = "ID_", insertable = false, updatable = false)
     val id: String,
 
-    @Column(name = "REV_")
+    @Column(name = "REV_", insertable = false, updatable = false)
     val revision: Int,
 
     @ManyToOne(fetch = FetchType.LAZY)
-    @JoinColumn(name = "ROOT_PROC_INST_ID_")
+    @JoinColumn(name = "ROOT_PROC_INST_ID_", insertable = false, updatable = false)
     val rootProcessInstance: CamundaExecution?,
 
     @ManyToOne(fetch = FetchType.LAZY)
-    @JoinColumn(name = "PROC_INST_ID_")
+    @JoinColumn(name = "PROC_INST_ID_", insertable = false, updatable = false)
     var processInstance: CamundaExecution?,
```

Comment:
`var` implies that it is mutable? Is there a reason why this remains `var` ?

Reply:
@sdegroot I could find similar 2 usages of this `var`:
`execution.setProcessInstance(execution);`

I guess that could be changed to a custom getter returning `this` when the value is `null`, but that's out of scope for this PR.

Reply:
I'd say that if you mark a class as `@Immutable`, it should be immutable :) 

This might cause problems later on. I can imagine hibernate is not happy if it notices that the property value is changed while it is set to `updatable = false`

Reply:
The annotation name is a bit unfortunate:
https://docs.jboss.org/hibernate/orm/5.2/javadocs/org/hibernate/annotations/Immutable.html

`An immutable entity may not be updated by the application. Updates to an immutable entity will be ignored, but no exception is thrown.`